src/components/oculus-touch-controls.js

```diff
@@ -139,55 +156,76 @@ module.exports.Component = registerComponent('oculus-touch-controls', {
       controller: 0,
       rotationOffset: data.rotationOffset !== -999 ? data.rotationOffset : isRightHand ? -90 : 90
     });
     this.updateControllerModel();
   },
 
+  addControllersUpdateListener: function () {
+    this.el.sceneEl.addEventListener('controllersupdated', this.onControllersUpdate, false);
+  },
+
+  removeControllersUpdateListener: function () {
+    this.el.sceneEl.removeEventListener('controllersupdated', this.onControllersUpdate, false);
+  },
+
   onControllersUpdate: function () {
-    if (!this.everGotGamepadEvent) { this.checkIfControllerPresent(); }
+    // Due to an apparent bug in FF Nightly
+    // where only one gamepadconnected / disconnected event is fired,
+    // which makes it difficult to handle in individual controller entities,
+    // we no longer remove the controllersupdate listener when we get a gamepad event.
+    this.checkIfControllerPresent();
   },
 
-  // currently, browser bugs prevent capacitive touch events from firing on trigger and grip;
-  // however those have analog values, and this (below button-down values) can be used to fake them
+  // Currently, browser bugs prevent capacitive touch events from firing on trigger and grip;
+  // however those have analog values, and this (below button-down values) can be used to fake them.
   isEmulatedTouchEvent: function (analogValue) {
     return analogValue && (analogValue >= EMULATED_TOUCH_THRESHOLD);
   },
 
   onButtonChanged: function (evt) {
-    var button = this.mapping[this.data.hand]['button' + evt.detail.id];
+    var button = this.mapping[this.data.hand].buttons[evt.detail.id];
     var buttonMeshes = this.buttonMeshes;
     var isPreviousValueEmulatedTouch;
     var analogValue;
     var isEmulatedTouch;
 
-    // at the moment, if trigger or grip,
-    // touch events aren't happening (touched is stuck true);
-    // synthesize touch events from very low analog values
-    if (button !== 'trigger' && button !== 'grip') { return; }
-    analogValue = evt.detail.state.value;
-    isPreviousValueEmulatedTouch = this.isEmulatedTouchEvent(this.previousButtonValues[button]);
-    this.previousButtonValues[button] = analogValue;
-    isEmulatedTouch = this.isEmulatedTouchEvent(analogValue);
-    if (isEmulatedTouch !== isPreviousValueEmulatedTouch) {
-      (isEmulatedTouch ? this.onButtonTouchStart : this.onButtonTouchEnd)(evt);
-    }
-    if (!buttonMeshes) { return; }
-    if (button === 'trigger' && buttonMeshes.trigger) {
-      buttonMeshes.trigger.rotation.x = -analogValue * (Math.PI / 24);
+    if (!button) { return; }
+
+    if (button === 'trigger' || button === 'grip') {
+      // At the moment, if trigger or grip,
+      // touch events aren't happening (touched is stuck true);
+      // synthesize touch events from very low analog values.
+      analogValue = evt.detail.state.value;
+      isPreviousValueEmulatedTouch = this.isEmulatedTouchEvent(this.previousButtonValues[button]);
```

Comment:
update: the 2017-03-22 build of Nightly makes Touch mapping compatible with Chromium, so #2446 is no longer required.  Also, Nightly provides analog value for all buttons, so we do in fact need this emulated touch code (e.g. to make hand-controls work correctly) since Nightly does not otherwise provide capacitive touch events.